core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala

```diff
@@ -29,25 +26,23 @@ import org.apache.kafka.common.errors.GroupMaxSizeReachedException
 import org.apache.kafka.common.protocol.ApiKeys
 import org.apache.kafka.common.requests.{FindCoordinatorRequest, FindCoordinatorResponse}
 import org.junit.Assert._
-import org.junit.{After, Before, Ignore, Test}
+import org.junit.{After, Ignore, Test}
 
 import scala.collection.JavaConverters._
-import scala.collection.mutable.ArrayBuffer
-import scala.concurrent.duration.Duration
-import scala.concurrent.{Await, ExecutionContext, ExecutionContextExecutor, Future => SFuture}
+import scala.collection.mutable
 
 /**
  * Integration tests for the consumer that cover basic usage as well as server failures
  */
-class ConsumerBounceTest extends BaseRequestTest with Logging {
-  val topic = "topic"
-  val part = 0
-  val tp = new TopicPartition(topic, part)
+class ConsumerBounceTest extends BaseConsumerTest with Logging {
```

Comment:
Extending `BaseConsumerTest` means we inherit its test cases (i.e. `testSimpleConsumption` and `testCoordinatorFailover`. I wonder if we should move these tests somewhere else.

Reply:
We have 6 test classes that make use of those test cases. Only `ConsumerBounceTest` is redundant. I don't think it hurts that much.

We could create a new class `BasicConsumerTests` which defines those tests if you think that's a good approach

Reply:
Yeah, it is not that big of a deal. Just it's a bit annoying to extend the build time even more because of tests we know are redundant. I think I slightly prefer your suggestion for `BasicConsumerTests`. Or maybe we can call the new class `BaseConsumerTest` and rename the existing one to `AbstractConsumerTest`?

The use of inheritance in these test cases has been a bit problematic in general. It would probably be better to split out any shared utility logic into a separate utility class. We don't have to do that here though.

Reply:
I like your thinking. Let's leave this as it is for the time being so we can merge this PR and reduce the flakiness of ConsumerBounceTest.

I've opened up a follow-up JIRA to track this - https://issues.apache.org/jira/browse/KAFKA-8176